appdatabase/database.go

```diff
@@ -76,6 +87,54 @@ func InitializeDB(path, password string, kdfIterationsNumber int) (*sql.DB, erro
 	return db, nil
 }
 
+func FixMissingKeyUIDForAccounts(sqlTx *sql.Tx) error {
+	rows, err := sqlTx.Query(`SELECT address,pubkey FROM accounts WHERE pubkey IS NOT NULL AND type != '' AND type != 'generated'`)
+	if err != nil {
+		log.Error("Migrating accounts: failed to query accounts", "err", err.Error())
+		return err
+	}
+	defer rows.Close()
+	for rows.Next() {
+		var address e_types.Address
+		var pubkey e_types.HexBytes
+		err = rows.Scan(&address, &pubkey)
+		if err != nil {
+			log.Error("Migrating accounts: failed to scan records", "err", err.Error())
+			return err
+		}
+		pk, err := crypto.UnmarshalPubkey(pubkey)
+		if err != nil {
+			log.Error("Migrating accounts: failed to unmarshal pubkey", "err", err.Error(), "pubkey", string(pubkey))
+			return err
+		}
+		pkBytes := sha256.Sum256(crypto.FromECDSAPub(pk))
+		keyUIDHex := hexutil.Encode(pkBytes[:])
+		_, err = sqlTx.Exec(`UPDATE accounts SET key_uid = ? WHERE address = ?`, keyUIDHex, address)
+		if err != nil {
+			log.Error("Migrating accounts: failed to update key_uid for imported accounts", "err", err.Error())
+			return err
+		}
+	}
+
+	var walletRootAddress e_types.Address
+	err = sqlTx.QueryRow(`SELECT wallet_root_address FROM settings WHERE synthetic_id='id'`).Scan(&walletRootAddress)
+	if err == sql.ErrNoRows {
+		// we shouldn't reach here, but if we do, it probably happened from the test
+		log.Warn("Migrating accounts: no wallet_root_address found in settings")
+		return nil
+	}
+	if err != nil {
+		log.Error("Migrating accounts: failed to get wallet_root_address", "err", err.Error())
+		return err
+	}
+	_, err = sqlTx.Exec(`UPDATE accounts SET key_uid = ?, derived_from = ? WHERE type = '' OR type = 'generated'`, CurrentAppDBKeyUID, walletRootAddress.Hex())
+	if err != nil {
+		log.Error("Migrating accounts: failed to update key_uid/derived_from", "err", err.Error())
+		return err
+	}
+	return nil
```

Comment:
This code doesn't appear to be ran during tests, which suggests that in the test case there are no settings present